source/extensions/tracers/opentelemetry/tracer.cc

```diff
@@ -122,8 +122,41 @@ void Span::setAttribute(absl::string_view name, const OTelAttribute& attribute_v
   *span_.add_attributes() = key_value;
 }
 
+::opentelemetry::proto::trace::v1::Status_StatusCode
+convertGrpcStatusToTraceStatusCode(::opentelemetry::proto::trace::v1::Span_SpanKind kind,
+                                   absl::string_view value) {
+  uint64_t grpc_status_code;
+  if (!absl::SimpleAtoi(value, &grpc_status_code)) {
+    return ::opentelemetry::proto::trace::v1::Status::STATUS_CODE_UNSET;
+  }
+
+  Grpc::Status::GrpcStatus grpc_status = static_cast<Grpc::Status::GrpcStatus>(grpc_status_code);
+  // Check mapping https://opentelemetry.io/docs/specs/semconv/rpc/grpc/#grpc-status
+  if (kind == ::opentelemetry::proto::trace::v1::Span::SPAN_KIND_SERVER) {
+    if (grpc_status == Grpc::Status::WellKnownGrpcStatus::Ok) {
+      return ::opentelemetry::proto::trace::v1::Status::STATUS_CODE_UNSET;
+    }
+    return ::opentelemetry::proto::trace::v1::Status::STATUS_CODE_ERROR;
+  }
```

Comment:
This code doesn't match the table at that URL. From a brief glance it seems that grpc-status Ok is always STATUS_CODE_UNSET and that a number of other grpc-status values are STATUS_CODE_UNSET for server spans, and are always STATUS_CODE_ERROR for client spans.

Reply:
Looking more closely, I think its just reversed. If kind == SPAN_KIND_CLIENT, check for ok and otherwise return error otherwise for server, use the switch below.